### PR TITLE
Replace depth mask with stencil mask

### LINKED ISSUE #528: Artemis minimap circle mask overrides Wynnlib side indicator
Besides being a conflict, this is also indicative that our depth mask code isn't well enough.
image

### PR DESCRIPTION
**Pull request by Incompleteusern**
Should fix #528

### CODE REVIEW

File: fabric/src/main/java/com/wynntils/fabric/mixins/RenderTargetMixin.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            method = "createBuffers",
+            at =
+                    @At(
+                            value = "INVOKE",
+                            target =
+                                    "Lcom/mojang/blaze3d/platform/GlStateManager;_texImage2D(IIIIIIIILjava/nio/IntBuffer;)V",
+                            ordinal = 0,
+                            remap = false))
+    public void init(Args args) {
+        if (RenderUtils.isStencilEnabled()) {
```
Comment (magicus) [inline fabric/src/main/java/com/wynntils/fabric/mixins/RenderTargetMixin.java]: I'm not really fond of this. I fear it has a high risk of conflict with other mods. Is there some other way you can go about to achieve what you need? (This goes for both `@ModifyArgs`)

File: quilt/src/main/resources/wynntils.mixins.quilt.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "client": [
+    "RenderTargetMixin"
+  ],
+  "compatibilityLevel": "JAVA_17",
```
Comment (magicus) [inline quilt/src/main/resources/wynntils.mixins.quilt.json:5]: What was the baseline before? Or is this specified in multiple places?

Comment (Incompleteusern) [reply quilt/src/main/resources/wynntils.mixins.quilt.json:5]: This is copy pasted from other places

**Comment by Incompleteusern** [comment]
image

**Comment by Incompleteusern** [comment]
Still needs testing that it works on quilt and forge (mainly forge).

File: fabric/src/main/java/com/wynntils/fabric/mixins/RenderTargetMixin.java (hunk 1 above)
Comment (Incompleteusern) [reply fabric/src/main/java/com/wynntils/fabric/mixins/RenderTargetMixin.java]: I doubt there will be much conflict, as I feel any other modification would be also for enabling stencil.

**Comment by kristofbolyai** [comment]
Forge works fine. 

Fabric/Quilt:
image

File: fabric/src/main/java/com/wynntils/fabric/mixins/RenderTargetMixin.java (hunk 1 above)
Comment (kristofbolyai) [reply fabric/src/main/java/com/wynntils/fabric/mixins/RenderTargetMixin.java]: We agreed to merge this on the basis that if this creates problems, we revert it

**Pull request merged**